lib/modules/dosomething/dosomething_user/dosomething_user.module

```diff
@@ -429,11 +433,55 @@ function dosomething_user_validate_address_field($form, &$form_state) {
     }
     // We have a full address, save it!
     else {
-      dosomething_user_set_address_values($form, $form_state, $formatted_address);
+      if ($form['#form_id'] === 'dosomething_user_info_form') {
+        dosomething_user_set_address_values_from_info_form($form, $form_state, $formatted_address);
+      }
+      else {
+        dosomething_user_set_address_values_from_edit_form($form, $form_state, $formatted_address);
+      }
     }
```

Comment:
@angaither I can totally change though to go about it a different route! Just not sure/don't understand how to follow the menu hook to validate the way JS does and I thought going about it this way was what you meant about creating a new validation/re-factoring?